src/GraphQL.Serialization.Newtonsoft/Http/DocumentWriter.cs

```diff
@@ -0,0 +1,49 @@
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace GraphQL.Http
+{
+    public class DocumentWriter : IDocumentWriter
+    {
+        private readonly JsonArrayPool _jsonArrayPool = new JsonArrayPool(ArrayPool<char>.Shared);
+        private readonly JsonSerializer _serializer;
+        internal static readonly Encoding Utf8Encoding = new UTF8Encoding(false);
+
+        public DocumentWriter()
+            : this(indent: false)
+        {
+        }
+
+        public DocumentWriter(bool indent)
+            : this(
+                indent ? Formatting.Indented : Formatting.None,
+                new JsonSerializerSettings())
+        {
+        }
+
+        public DocumentWriter(Formatting formatting, JsonSerializerSettings settings)
+        {
+            _serializer = JsonSerializer.CreateDefault(settings);
+            _serializer.Formatting = formatting;
+            _serializer.ContractResolver = new NewtonsoftContractResolver(_serializer.ContractResolver);
```

Comment:
```c#
_serializer = JsonSerializer.CreateDefault(settings);
_serializer.Formatting = formatting;
_serializer.ContractResolver = new NewtonsoftContractResolver(_serializer.ContractResolver);
```
->
```c#
if (_settings.ContractResolver == null)
   _settings.ContractResolver =  new NewtonsoftContractResolver()
_serializer = JsonSerializer.CreateDefault(settings);
 _serializer.Formatting = formatting;
```

Reply:
By the way, this way we allow the client to use their own serialization format.

Reply:
I see your point.  I didn't want to modify the incoming settings object as it could be a static/shared object used multiple times in various places, and we shouldn't modify it.

Reply:
Your comment makes sense. Then we just need to remove the `formatting` parameter.